docs/specs/servers/blockchain/blockchain-permissions-api.md

```diff
@@ -0,0 +1,174 @@
+# Blockchain API Sessions and Permissions
+
+## Sessions permissions storage
+
+### Get permissions list for account
+
+Used to get account list of active sessions
+
+`GET /v1/sessions/{address}?projectId={projectId}`
+
+* `address` - CAIP-10 address format.
+* `projectId` - Required. The project identifier.
+
+#### Success response body:
+
+```typescript
+{
+    pci: string[]
+}
+```
+
+* `pci` - List of sessions PCIs for this account.
+
+#### Response error codes:
+
+* `400 Bad request` - Wrong requested address format.
+
+### Get permission by PCI
+
+Used to get permission by PCI
+
+`GET /v1/sessions/{address}/{pci}?projectId={projectId}`
+
+* `address` - CAIP-10 address format.
+* `pci` - Permission identifier.
+* `projectId` - Required. The project identifier.
+
+#### Success response body:
+
+```typescript
+{
+    "permissionType": string,
+    "data": string,
+    "required": boolean,
+    "onChainValidated": boolean
+}
+```
+
+#### Response error codes:
+
+* `400 Bad request` - Wrong requested address format or PCI not found.
+
+## Add a new permission 
+
+Creating a new permission session for the account
+
+`POST /v1/sessions/{address}?projectId={projectId}`
+
+* `address` - CAIP-10 address format.
+* `projectId` - Required. The project identifier.
+
+### Request body:
+
+The POST request body should be in JSON format and following schema:
+
+```typescript
+{
+    permission:
+        {
+          "permissionType": string,
+          "data": string,
+          "required": boolean,
+          "onChainValidated": boolean
+        }
+}
+```
+
+#### Success response body:
+
+Response will contain a new generated key and PCI of the new permission.
+
+```typescript
+{
+    pci: string,
+    key: string
+}
+```
+
+* `pci` - New unique permission controller identifier.
+* `key` - Generated signing (private) ECDSA P256 key in DER, SEC1 format encoded by Base64.
+
+#### Response error codes:
+
+* `400 Bad request` - Wrong format in request.
+
+## Updating permissions context
+
+Updating permissions context for the certain permission idenitifier.
+
+`POST /v1/sessions/{address}/context?projectId={projectId}`
+
+* `address` - CAIP-10 address format.
+* `projectId` - Required. The project identifier.
+
+### Request body:
+
+The POST request body should be in JSON format and following schema based on the [ERC-7715](https://github.com/ethereum/ERCs/blob/a75e2d80698441f5da9e0fe98d38122a862aed89/ERCS/erc-7715.md#signers):
+
+```typescript
+{
+    pci: string,
+    signature: string,
+    context: {
+      signer: {
+        type: string,
+        data:{
+          ids: string[],
+        }
+      },
+      expiry: number,
+      signerData: {
+        userOpBuilder: string
+      },
+      factory: string,
+      factoryData: string,
+      permissionsContext: string
+    }
+}
+```
```

Comment:
`context` field should contain the complete response of `wallet_grantPermission` request received by the Dapp/client from wallet, because it contains list of `granted permissions` for requested session. The initial set of permissions stored by the co-signer could be different than final granted permissions.

Reply:
Do you mean the context should be the list of [granted permissions](https://github.com/ethereum/ERCs/pull/436/files#diff-30a722e02b85e6e33ea9d633379985173e9a669ffffdbba3662e28c0ec8adbe7R200) from ERC7715?